### PR TITLE
[Feature][task-plugins-12820] azure azure-sql datasource plugins

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/12820
supports 5 auth mode to azure sql

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-azure-sql/src/main/java/org/apache/dolphinscheduler/plugin/datasource/azuresql/param/AzureSQLDataSourceProcessor.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        AzureSQLDataSourceParamDTO azureSQLDatasourceParamDTO = new AzureSQLDataSourceParamDTO();
+        azureSQLDatasourceParamDTO.setDatabase(connectionParams.getDatabase());
+        azureSQLDatasourceParamDTO.setUserName(connectionParams.getUser());
+        azureSQLDatasourceParamDTO.setOther(connectionParams.getOther());
+        azureSQLDatasourceParamDTO.setPort(Integer.parseInt(hostPortArray[0].split(Constants.COLON)[1]));
+        azureSQLDatasourceParamDTO.setHost(hostPortArray[0].split(Constants.COLON)[0]);
+        azureSQLDatasourceParamDTO.setMode(connectionParams.getMode());
+        switch (azureSQLDatasourceParamDTO.getMode()) {
+            case AD_MSI:
+                if (!StringUtils.isEmpty(connectionParams.getMSIClientId())) {
```
Comment (caishunfeng) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-azure-sql/src/main/java/org/apache/dolphinscheduler/plugin/datasource/azuresql/param/AzureSQLDataSourceProcessor.java]:
```suggestion
                if (StringUtils.isNotEmpty(connectionParams.getMSIClientId())) {
```

**Review by caishunfeng — DISMISSED** [review]
LGTM, some nip

**Comment by caishunfeng** [comment]
@Tianqi-Dotes please add the UI and doc too.

**Comment by caishunfeng** [comment]
image

Need to do code format.

**Comment by Tianqi-Dotes** [comment]
> @Tianqi-Dotes please add the UI and doc too.
UI PR:
https://github.com/apache/dolphinscheduler/pull/13200
DOC will be submit another PR

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-azure-sql/src/main/java/org/apache/dolphinscheduler/plugin/datasource/azuresql/param/AzureSQLDataSourceProcessor.java (hunk 1 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-azure-sql/src/main/java/org/apache/dolphinscheduler/plugin/datasource/azuresql/param/AzureSQLDataSourceProcessor.java]: done

**Review by caishunfeng — APPROVED** [review]

**Pull request merged**